pkg/controller.v1/common/job.go

```diff
@@ -307,9 +307,24 @@ func (jc *JobController) ReconcileJobs(
 				MinResources:      minResources,
 			}
 
-			_, err := jc.SyncPodGroup(metaObject, pgSpec)
+			syncReplicas := true
+			pg, err := jc.SyncPodGroup(metaObject, pgSpec)
 			if err != nil {
 				log.Warnf("Sync PodGroup %v: %v", jobKey, err)
+				syncReplicas = false
+			}
+
+			// Delay pods creation until podgroup status is inqueue
+			if pg == nil || pg.Status.Phase == "" || pg.Status.Phase == v1beta1.PodGroupPending {
+				log.Warnf("PodGroup %v unschedulable", jobKey)
+				syncReplicas = false
+			}
+
+			if !syncReplicas {
+				now := metav1.Now()
+				jobStatus.LastReconcileTime = &now
+
+				return jc.Controller.UpdateJobStatusInApiServer(job, &jobStatus)
```

Comment:
I think we could add some comments here to let us know that we do this update to trigger another reconcile. And maybe in the future the operator should watch PodGroup directly.